core/src/services/routing.js

```diff
@@ -15,6 +15,26 @@ const getViewUrl = pathData => {
   return lastElement ? lastElement.viewUrl : '';
 };
 
+const getDefaultPathSegment = function(pathData) {
```

Comment:
I would propose to refactor the function to the following:
```
const getDefaultPathSegment = function(pathData) {
  const lastElement = [...pathData.navigationPath].pop();
  const pathExists = lastElement.children.find(
    childNode => childNode.pathSegment === lastElement.defaultPathSegment
  );
  if (lastElement.defaultPathSegment && pathExists) {
    return lastElement.defaultPathSegment;
  } else {
    return lastElement.children[0].pathSegment;
  }
};
```

It is a little less efficient because you will calculate always pathExists, but the code is imho much more readable and it pays off